app/react/translations/edit/editors/ArrayEditor/index.js

```diff
@@ -68,45 +74,52 @@ export default class ArrayEditor extends React.PureComponent {
   handleKeyDown = (event: InputEvent, { index, length }: ArrayInfo) => {
     const isEmpty: boolean = event.currentTarget && !event.currentTarget.value;
     const isLast: boolean = index === length - 1;
+
     if (event.keyCode === 13) {
       this.pushArray();
     }
     if (event.keyCode === 8 && isLast && isEmpty) {
       event.preventDefault();
       this.popArray();
     }
+    if (event.keyCode === 9 && isLast) {
+      this.handleSubmit();
```

Comment:
Call it with event `this.handleSubmit(event)` and then you can skip check for event in `handleSubmit` method.